src/preface.md

```diff
@@ -1 +1,68 @@
+
+[prerequisites]: #scope-of-this-book
+[resources]: #additional-resources
+
+## Scope of This Book
+
+In this book, we focus specifically on the `esp-rs` driver and its ecosystem. It does **not** serve as:
+
+- A general Rust programming language tutorial: if you are new to Rust, we recommend starting with [The Rust Programming Language][rust-book].
+- An embedded development guide: prior familiarity with basic embedded development is assumed.
+- A collection of examples: This book explains the overall process of working with the `esp-rs` ecosystem. More examples can be found [here][examples] or in the [Awesome ESP Rust][awesome-esp-rust] repository. 
+
+[examples]: https://github.com/esp-rs/esp-hal/tree/main/examples
+
+## Stability and Availability
+
+The `esp-rs` driver is under active development and may be subject to changes as it evolves. While we strive for stability, users should expect periodic modifications as we improve the API, improve performance, and introduce new features. Those [modules] that are already stabilised will not be subject to frequent dramatic changes, while the `unstable` parts of the driver are being actively worked on and will likely undergo changes in future versions of the driver. 
+
+Additionally, the ecosystem of Rust itself, in a broader meaning, is still in its infancy. Some dependencies, tools, or language features may change over time. We encourage developers to stay up to date with the latest releases and participate in discussions to help shape the future of Rust on Espressif platforms.
+
+[modules]: https://docs.espressif.com/projects/rust/esp-hal/1.0.0-beta.0/esp32c6/esp_hal/index.html#modules
+
+## Additional Resources
+
+If you're unfamiliar with certain concepts covered in this book or would like to deepen your understanding, the following resources may be helpful:
+
+| Resource                                               | Description                                                                          |
+| ------------------------------------------------------ | ------------------------------------------------------------------------------------ |
+| [The Rust Programming Language][rust-book]             | Learn Rust fundamentals before diving into embedded development.                     |
+| [The Embedded Rust Book][embedded-rust-book]           | A collection of resources from Rust's Embedded Working Group.                        |
+| [The Embedonomicon][embedonomicon]                     | Detailed insights into low-level embedded Rust programming.                          |
+| [Embedded Rust (std) on Espressif][std-training]       | Getting started guide for using Rust's `std` environment on Espressif SoCs.         |
+| [Embedded Rust (no_std) on Espressif][no_std-training] | Guide for working in `no_std` environments with Espressif SoCs.                     |
```

Comment:
Id recommend the no_std training before the std_training

Reply:
Didn't we decide we weren't going to talk about `std` at all, since it's not supported by us and is probably just confusing?